pkg/antctl/raw/check/cluster/command.go

```diff
@@ -0,0 +1,210 @@
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/ptr"
+
+	"antrea.io/antrea/pkg/antctl/raw/check"
+)
+
+func Command() *cobra.Command {
+	o := newOptions()
+	command := &cobra.Command{
+		Use:   "cluster",
+		Short: "Runs pre installation checks",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return Run(o)
+		},
+	}
+	command.Flags().StringVarP(&o.antreaNamespace, "namespace", "n", o.antreaNamespace, "Configure Namespace in which Antrea is running")
+	return command
+}
+
+type options struct {
+	antreaNamespace string
+}
+
+func newOptions() *options {
+	return &options{
+		antreaNamespace: "kube-system",
+	}
+}
+
+const (
+	antreaNamespace = "kube-system"
+	testNamespace   = "antrea-test"
+	deploymentName  = "check-cluster"
+	podReadyTimeout = 1 * time.Minute
+)
+
+type Test interface {
+	Run(ctx context.Context, testContext *testContext) error
+}
+
+var testsRegistry = make(map[string]Test)
+
+func RegisterTest(name string, test Test) {
+	testsRegistry[name] = test
+}
+
+type testContext struct {
+	client          kubernetes.Interface
+	config          *rest.Config
+	clusterName     string
+	antreaNamespace string
+	namespace       string
+}
+
+func Run(o *options) error {
+	client, config, clusterName, err := check.NewClient()
+	if err != nil {
+		return fmt.Errorf("unable to create Kubernetes client: %s", err)
+	}
+	ctx := context.Background()
+	testContext := NewTestContext(client, config, clusterName, o)
+	if err := testContext.setup(ctx); err != nil {
+		return err
+	}
+	var numSuccess, numFailure int
+	for name, test := range testsRegistry {
+		testContext.Header("Running test: %s", name)
+		if err := test.Run(ctx, testContext); err != nil {
+			testContext.Fail("Test %s failed: %v", name, err)
+			numFailure++
+		} else {
+			testContext.Success("Test %s passed", name)
+			numSuccess++
+		}
+	}
+	testContext.Log("Test finished: %v tests succeeded, %v tests failed ", numSuccess, numFailure)
+	check.Teardown(ctx, testContext.client, testContext.clusterName, testContext.namespace)
+	if numFailure > 0 {
+		return fmt.Errorf("%v/%v tests failed", numFailure, len(testsRegistry))
+	}
+	return nil
+}
+
+func (t *testContext) setup(ctx context.Context) error {
+	t.Log("Creating Namespace %s for pre installation tests...", t.namespace)
+	_, err := t.client.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: t.namespace}}, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to create Namespace %s: %s", t.namespace, err)
+	}
+	deployment := check.NewDeployment(check.DeploymentParameters{
+		Name:        deploymentName,
+		Image:       "antrea/antrea-agent-ubuntu:latest",
+		Replicas:    1,
+		Command:     []string{"sleep", "infinity"},
```

Comment:
This will ignore the SIGTERM from the container runtime. As a result we have to wait for the 30s timeout before the Pod is deleted and clean-up takes a long time. You should try something like this:
```
Command:  []string{"bash", "-c"},
// Make sure that we exit immediately when SIGTERM is received.
Args:        []string{"trap 'exit 0' SIGTERM; sleep infinity & pid=$!; wait $pid"},
```